### ISSUE TITLE
Lots of "Element query executed before Craft is fully initialized." in my logs

### ISSUE THREAD

**Issue by thomasjonas** [engram-design/Workflow#200]
### Describe the bug

In my Craft CMS's logs I see lots (and really, lots...) of warnings saying `Element query executed before Craft is fully initialized.` The stack trace is pointing me to `/vendor/verbb/workflow/src/Workflow.php(77): yii\web\User->getIdentity()`

I've also found the [following issue on Craft CMS](https://github.com/craftcms/cms/issues/11033).

Maybe it would work by not setting `hasCpSection` in the init function, but using the `EVENT_REGISTER_CP_NAV_ITEMS` hook. Something like this:

```
Event::on(
            Cp::class,
            Cp::EVENT_REGISTER_CP_NAV_ITEMS,
            function (RegisterCpNavItemsEvent $event) {
                // Only show the menu item if user has permission to overview
                if ($currentUser = Craft::$app->getUser()->getIdentity()) {
                    if ($currentUser->can('workflow-overview')) {
                        $event->navItems[] = [
                            'url' => 'workflow',
                            'label' => 'Workflow',
                            'icon' => '@workflow/icon.svg',
                        ];
                    }
                }
            }
        );
```

### Steps to reproduce

1. Install and enable the workflow plugin
2. Make a request on the Craft CMS admin panel
3. Look at Craft's web logs 

### Craft CMS version

4.5.14

### Plugin version

2.0.7

### Multi-site?

No

**Comment by engram-design** [comment]
Fixed for the next release. To get this early, run `composer require verbb/workflow:"dev-craft-4 as 2.0.7"`.

**Comment by thomasjonas** [comment]
Great! Thanks for the quick fix!

**Comment by engram-design** [comment]
Fixed in [2.0.8](https://github.com/verbb/workflow/releases/tag/2.0.8)
